pkg/console/operator/sync_v400.go

```diff
@@ -392,6 +403,86 @@ func (co *consoleOperator) SyncRoute(operatorConfig *operatorv1.Console) (*route
 	return rt, rtIsNew, rtErr
 }
 
+func (c *consoleOperator) SyncCustomLogoConfigMap(operatorConfig *operatorsv1.Console) error {
+	// validate first, to avoid a broken volume mount & a crashlooping console
+	okToMount, err := c.ValidateCustomLogo(operatorConfig)
+
+	if okToMount {
+		if err := c.UpdateCustomLogoSyncSource(operatorConfig); err != nil {
+			return err
+		}
+	}
+	return err
+}
+
+// on each pass of the operator sync loop, we need to check the
+// operator config for a custom logo.  If this has been set, then
+// we notify the resourceSyncer that it needs to start watching this
+// configmap in its own sync loop.  Note that the resourceSyncer's actual
+// sync loop will run later.  Our operator is waiting to receive
+// the copied configmap into the console namespace for a future
+// sync loop to mount into the console deployment.
+func (c *consoleOperator) UpdateCustomLogoSyncSource(operatorConfig *operatorsv1.Console) error {
+	source := resourcesynccontroller.ResourceLocation{}
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+
+	if logoConfigMapName != "" {
+		source.Name = logoConfigMapName
+		source.Namespace = api.OpenShiftConfigNamespace
+	}
+	// if no custom logo provided, sync an empty source to delete
+	return c.resourceSyncer.SyncConfigMap(
+		resourcesynccontroller.ResourceLocation{Namespace: api.OpenShiftConsoleNamespace, Name: api.OpenShiftCustomLogoConfigMapName},
+		source,
+	)
+	return nil
+}
+
+func (c *consoleOperator) ValidateCustomLogo(operatorConfig *operatorsv1.Console) (okToMount bool, invalid error) {
+	logoConfigMapName := operatorConfig.Spec.Customization.CustomLogoFile.Name
+	logoImageKey := operatorConfig.Spec.Customization.CustomLogoFile.Key
+
+	// if unset, allow the sync to set a null source
+	if len(logoConfigMapName) == 0 && len(logoImageKey) == 0 {
+		return true, nil
+	}
+
+	// both set, or both unset
+	if (len(logoConfigMapName) == 0) != (len(logoImageKey) == 0) {
+		return false, customerrors.NewCustomLogoError("custom logo file name and key must both be set")
+	}
+
+	// nothing to do if there is no name. return
+	if len(logoConfigMapName) == 0 {
+		// TODO: degraded, error?
+		return false, nil
+	}
+
+	// now check the actual configmap...
+	logoConfigMap, err := c.configMapClient.ConfigMaps(api.OpenShiftConfigNamespace).Get(logoConfigMapName, metav1.GetOptions{})
+	// If we 404, the logo file may not have been created yet.
+	// TODO: should we err out then?
+	if err != nil {
+		return false, customerrors.NewCustomLogoError(fmt.Sprintf("custom logo file %v not found", logoConfigMapName))
+	}
+
+	logoBytes := logoConfigMap.BinaryData[logoImageKey]
+	// image headers are at least 10 bytes, so if its less than this it is broken
```

Comment:
Do we have to add a check ? or will ad it in the future to the `IsLikelyCommonImageFormat()` ?

Reply:
Thx, I will eliminate this comment.  It was a thought to be simplistic about the check.